src/charm.py

```diff
@@ -520,6 +547,11 @@ def _on_set_password(self, event: ActionEvent) -> None:
             return
 
         new_password = event.params.get(Config.Actions.PASSWORD_PARAM_NAME, generate_password())
+        if len(new_password) > Config.Secrets.MAX_PASSWORD_LENGTH:
+            event.fail(
+                f"Password cannot be longer than {Config.Secrets.MAX_PASSWORD_LENGTH} characters."
+            )
+            return
```

Comment:
Is there a lower limit, too? Hmm, on that note: if we pass an empty string value for a Juju Secret, the operation will fail.

Reply:
Lower limit will be introduced later for both charms. The upper limit set to avoid crash. 😄